src/components/App.tsx

```diff
@@ -29,66 +36,145 @@ class App extends React.Component<{}, State> {
+        }, hideDelay);
+    }
+
+    onAuthAction = (currentUser: GoogleUser) => (signedIn: boolean) => {
+        const userValid = validUser(currentUser);
+        if (signedIn && userValid) {
+            const userEmail = currentUser.getBasicProfile().getEmail();
+            this.setState({ isLoading: false, userEmail });
+        } else if (!userValid) {
+            if (signedIn) {
                 gapi.auth2.getAuthInstance().signOut();
+            } else {
+                this.showBanner("You need a TechSee email to use this app", 'alert-danger');
             }
         }
     }
 
-    renderBanner = (shouldRenderMenu: boolean) => {
-        if(!shouldRenderMenu) {
-            return (
-                <StatusBanner 
-                    mounted={this.state.showBanner} 
-                    success={false} 
-                    message={this.bannerMessage}
-                />
-            );
+    onEmployeeEnter = async () => {
+        const {inOffice, userEmail} = this.state;
+        try {
+            if(!inOffice) {
+                await workLogger.post("/log", {
+                    dateString: new Date().toString(),
+                    userEmail
+                });
+                this.setState({
+                    inOffice: true
+                });
+                this.showBanner("Swiped in successfuly");
+            } else {
+                this.showBanner("Tried to swipe in while already swiped");
+            }
+        } catch(err) {
+            this.showBanner("Failed to swipe in", 'alert-danger');
+            console.error(err);
         }
     }
 
-    renderContents = (shouldRenderMenu: boolean) => {
-        if(shouldRenderMenu) {
-            return <WorkLoggerMenu userEmail={this.state.userEmail}/>;
+    onEmployeeLeave = async () => {
+        const {inOffice, userEmail} = this.state;
+        try{
+            if(inOffice) {
+                await workLogger.put("/log", {
+                    dateString: new Date().toString(),
+                    userEmail
+                });
+                this.setState({
+                    inOffice: false
+                })
+                this.showBanner('Swiped out successfuly');
+            } else {
+                this.showBanner("Can't exit without entering");
+            }
+        } catch(err) {
+            if(err.message === "Network Error") {
+                this.showBanner("Failed to swipe out", 'alert-danger');
+                console.error(err);
+            }
+        }
+    }
+
+    onRequestLog = async () => {
+        const {userEmail} = this.state;
+        try {
+            await workLogger.post("/send", {
+                dateString: new Date().toString(),
+                userEmail
+            });
+            this.showBanner('Log sent successfuly')
+        } catch(err) {
+            this.showBanner("Failed to send log", 'alert-danger');
+            console.error(err);
         }
-        return <GoogleAuth/>;
     }
```

Comment:
Hoisted all API calls up to the top level component

Reply:
Ahhh so hoisted is basically lifting state. 
It makes much more sense for the App component to be handling all the api mumbo jumbo, thanks for clearing that up!

Reply:
![image](https://user-images.githubusercontent.com/18609911/66805017-a063b800-ef2c-11e9-95ec-cccd03bf2fa8.png)